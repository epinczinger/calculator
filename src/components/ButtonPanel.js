import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = props => {
  const blueButton = 'bg-blue-700 hover:bg-blue-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800';
  const greenButton = 'bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800';
  const greenLgButton = 'bg-green-700 hover:bg-green-800 flex justify-center w-32 h-16 lg:w-48 lg:h-20 border rounded border-gray-800';
  const yellowButton = 'bg-yellow-500 hover:bg-yellow-600 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800';

  return (
    <div className="bg-gray-800 text-lg md:text-xl">
      <div className="flex">
        <div className={blueButton}>
          <Button clickHandler={() => props.clickHandler('AC')} name="AC" />
        </div>
        <div className="bg-blue-700 hover:bg-blue-800 w-1/4 flex justify-center border rounded border-gray-800 min-w-12">
          <Button clickHandler={() => props.clickHandler('+/-')} name="+/-" />
        </div>
        <div className={blueButton}>
          <Button clickHandler={() => props.clickHandler('%')} name="%" />
        </div>
        <div className={yellowButton}>
          <Button clickHandler={() => props.clickHandler('รท')} name="รท" />
        </div>
      </div>
      <div className="flex">
        <div className={greenButton}>
          <Button clickHandler={() => props.clickHandler('7')} name="7" />
        </div>
        <div className={greenButton}>
          <Button clickHandler={() => props.clickHandler('8')} name="8" />
        </div>
        <div className={greenButton}>
          <Button clickHandler={() => props.clickHandler('9')} name="9" />
        </div>
        <div className={yellowButton}>
          <Button clickHandler={() => props.clickHandler('x')} name="x" />
        </div>
      </div>
      <div className="flex">
        <div className={greenButton}>
          <Button clickHandler={() => props.clickHandler('4')} name="4" />
        </div>
        <div className={greenButton}>
          <Button clickHandler={() => props.clickHandler('5')} name="5" />
        </div>
        <div className={greenButton}>
          <Button clickHandler={() => props.clickHandler('6')} name="6" />
        </div>
        <div className={yellowButton}>
          <Button clickHandler={() => props.clickHandler('-')} name="-" />
        </div>
      </div>
      <div className="flex">
        <div className={greenButton}>
          <Button clickHandler={() => props.clickHandler('1')} name="1" />
        </div>
        <div className={greenButton}>
          <Button clickHandler={() => props.clickHandler('2')} name="2" />
        </div>
        <div className={greenButton}>
          <Button clickHandler={() => props.clickHandler('3')} name="3" />
        </div>
        <div className={yellowButton}>
          <Button clickHandler={() => props.clickHandler('+')} name="+" />
        </div>
      </div>
      <div className="flex">
        <div className={greenLgButton}>
          <Button clickHandler={() => props.clickHandler('0')} name="0" />
        </div>
        <div className={greenButton}>
          <Button clickHandler={() => props.clickHandler('.')} name="." />
        </div>
        <div className={yellowButton}>
          <Button clickHandler={() => props.clickHandler('=')} name="=" />
        </div>
      </div>
    </div>
  );
};

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};

export default ButtonPanel;
