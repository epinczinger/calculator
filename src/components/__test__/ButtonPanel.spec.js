import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPanel from '../ButtonPanel';

describe('ButtonPannel component', () => {
  it('renders as expected matching snapshot', () => {
    const buttonPanel = renderer.create(<ButtonPanel clickHandler={Function} />).toJSON();
    expect(buttonPanel).toMatchSnapshot();
  });
});
