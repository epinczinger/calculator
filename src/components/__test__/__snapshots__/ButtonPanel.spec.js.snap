// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPannel component renders as expected matching snapshot 1`] = `
<div
  className="bg-gray-800 text-lg md:text-xl"
>
  <div
    className="flex"
  >
    <div
      className="bg-blue-700 hover:bg-blue-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
    </div>
    <div
      className="bg-blue-700 hover:bg-blue-800 w-1/4 flex justify-center border rounded border-gray-800 min-w-12"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
    </div>
    <div
      className="bg-blue-700 hover:bg-blue-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
    </div>
    <div
      className="bg-yellow-500 hover:bg-yellow-600 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
  </div>
  <div
    className="flex"
  >
    <div
      className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
    </div>
    <div
      className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
    </div>
    <div
      className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
    </div>
    <div
      className="bg-yellow-500 hover:bg-yellow-600 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
  </div>
  <div
    className="flex"
  >
    <div
      className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
    </div>
    <div
      className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
    </div>
    <div
      className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
    </div>
    <div
      className="bg-yellow-500 hover:bg-yellow-600 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
  </div>
  <div
    className="flex"
  >
    <div
      className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
    <div
      className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </div>
    <div
      className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
    <div
      className="bg-yellow-500 hover:bg-yellow-600 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
  </div>
  <div
    className="flex"
  >
    <div
      className="bg-green-700 hover:bg-green-800 flex justify-center w-32 h-16 lg:w-48 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
    </div>
    <div
      className="bg-green-700 hover:bg-green-800 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
    <div
      className="bg-yellow-500 hover:bg-yellow-600 flex justify-center w-16 h-16 lg:w-24 lg:h-20 border rounded border-gray-800"
    >
      <button
        className="w-full h-full"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
